js/controller.js

```diff
@@ -1163,6 +1165,27 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       this.mb.publish(OO.EVENTS.DISCOVERY_API.SEND_DISPLAY_EVENT, eventData);
     },
 
+    toggleVideoQualityPopOver: function() {
+      this.state.videoQualityOptions.showVideoQualityPopover = !this.state.videoQualityOptions.showVideoQualityPopover;
+      this.renderSkin();
+    },
+
+    receiveVideoQualityChangeEvent: function(eventName, targetBitrate) {
+        this.state.videoQualityOptions.selectedBitrate = {
+        "id": targetBitrate
+      };
+      this.renderSkin({
+          "videoQualityOptions": {
+            "availableBitrates": this.state.videoQualityOptions.availableBitrates,
+            "selectedBitrate": this.state.videoQualityOptions.selectedBitrate,
+            "showVideoQualityPopover":this.state.videoQualityOptions.showVideoQualityPopover
+          }
+        });
+      if(this.state.videoQualityOptions.showVideoQualityPopover == true) {
+        this.toggleVideoQualityPopOver();
+      }
```

Comment:
Second thing, why the new event and not using SET_TARGET_BITRATE? I don't see any difference in how the events are being differentiated.